### PR TITLE
limit super user to normal access on record changes

### PR DESCRIPTION
**Pull request by rebstar6**
Fixes #.

Changes in this pull request:
-
-

### CODE REVIEW

File: modules/api/src/main/scala/vinyldns/api/domain/AccessValidations.scala [hunk 1]
```diff
@@ -183,12 +183,12 @@ object AccessValidations extends AccessValidationAlgebra {
...
       recordOwnerGroupId: Option[String] = None): AccessLevel = auth match {
     case testUser if testUser.isTestUser && !zone.isTest => AccessLevel.NoAccess
-    case admin if admin.canEditAll || admin.isGroupMember(zone.adminGroupId) =>
+    case admin if admin.isGroupMember(zone.adminGroupId) =>
       AccessLevel.Delete
     case recordOwner
         if zone.shared && sharedRecordAccess(recordOwner, recordType, recordOwnerGroupId) =>
       AccessLevel.Delete
-    case supportUser if supportUser.canReadAll =>
+    case support if support.canReadAll =>
```
Comment (mitruly) [inline modules/api/src/main/scala/vinyldns/api/domain/AccessValidations.scala]: Should we also add `isSuper` here to give at least `Read` access? My understanding is that super users are still between support users and admins, with the added ability to choose owner groups that they do not belong to.

Comment (nimaeskandary) [reply modules/api/src/main/scala/vinyldns/api/domain/AccessValidations.scala]: can read all checks for the super and support flag

Comment (nimaeskandary) [reply modules/api/src/main/scala/vinyldns/api/domain/AccessValidations.scala]: could maybe rename to supportOrSuper

Comment (mitruly) [reply modules/api/src/main/scala/vinyldns/api/domain/AccessValidations.scala]: Good call, think renaming would make it clearer, but I'm good with this.

**Comment by mitruly** [comment]
Seems like there are a couple of failing tests:
```
- should update relative NS record without trailing dot *** FAILED *** (64 milliseconds)
[info]   java.util.NoSuchElementException: None.get
[info]   at scala.None$.get(Option.scala:366)
[info]   at scala.None$.get(Option.scala:364)
[info]   at vinyldns.api.ResultHelpers.rightValue(ResultHelpers.scala:68)
[info]   at vinyldns.api.ResultHelpers.rightValue$(ResultHelpers.scala:68)
[info]   at vinyldns.api.domain.record.RecordSetServiceIntegrationSpec.rightValue(RecordSetServiceIntegrationSpec.scala:43)
[info]   at vinyldns.api.domain.record.RecordSetServiceIntegrationSpec.$anonfun$new$16(RecordSetServiceIntegrationSpec.scala:348)
[info]   at org.scalatest.concurrent.Futures.whenReady(Futures.scala:619)
[info]   at org.scalatest.concurrent.Futures.whenReady$(Futures.scala:617)
[info]   at vinyldns.api.domain.record.RecordSetServiceIntegrationSpec.whenReady(RecordSetServiceIntegrationSpec.scala:43)
[info]   at vinyldns.api.domain.record.RecordSetServiceIntegrationSpec.$anonfun$new$15(RecordSetServiceIntegrationSpec.scala:347)
[info]   ...
[info] - should delete successfully for super user in shared zone *** FAILED *** (19 milliseconds)
[info]   'Left(vinyldns.api.domain.zone.NotAuthorizedError: User super does not have access to delete shared-record.shared-zone-test.)' was not an Right, but should have been. (RecordSetServiceIntegrationSpec.scala:551)
```

**Pull request merged**